dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -0,0 +1,118 @@
+        dataReader.ReadBytes(m_backgroundPayload);
+    }
+
+    PushNotificationReceivedEventArgs::PushNotificationReceivedEventArgs(winrt::PushNotificationReceivedEventArgs const& args)
+    {
+        m_args = args;
+    }
+
+    winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs PushNotificationReceivedEventArgs::CreateFromBackgroundTaskInstance(winrt::IBackgroundTaskInstance const& backgroundTask)
+    {
+        return winrt::make<PushNotificationReceivedEventArgs>(backgroundTask);
+    }
+
+    winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs PushNotificationReceivedEventArgs::CreateFromPushNotificationReceivedEventArgs(winrt::PushNotificationReceivedEventArgs const& args)
+    {
+        return winrt::make<PushNotificationReceivedEventArgs>(args);
+    }
+
+    winrt::com_array<uint8_t> PushNotificationReceivedEventArgs::Payload()
+    {
+        auto lock = m_lock.lock_shared();
+        winrt::com_array<uint8_t> payload{};
+        if (m_args)
+        {
+            RawNotification rawNotification = m_args.RawNotification();
+
+            IBuffer rawContentAsBuffer = rawNotification.ContentBytes();
+            DataReader dataReader = DataReader::FromBuffer(rawContentAsBuffer);
+
+            payload = winrt::com_array<uint8_t>(rawContentAsBuffer.Length());
+            dataReader.ReadBytes(payload);
+        }
+        else if (m_backgroundTaskInstance)
+        {
+            return winrt::com_array<uint8_t>(m_backgroundPayload.begin(), m_backgroundPayload.end());
+        }
+        return payload;
+    }
+
+    winrt::BackgroundTaskDeferral PushNotificationReceivedEventArgs::GetDeferral()
+    {
+        auto lock = m_lock.lock_shared();
+        if (m_backgroundTaskInstance)
+        {
+            return m_backgroundTaskInstance.GetDeferral();
+        }
+        return nullptr;
+    }
+
+    winrt::event_token PushNotificationReceivedEventArgs::Canceled(winrt::BackgroundTaskCanceledEventHandler const& handler)
+    {
+        {
+            auto lock = m_lock.lock_shared();
+            if (m_backgroundTaskInstance)
+            {
+                return m_backgroundTaskInstance.Canceled(handler);
+            }
+        }
+
+        return winrt::event_token{};
+    }
+    void PushNotificationReceivedEventArgs::Canceled(winrt::event_token const& token) noexcept
+    {
+        auto lock = m_lock.lock_shared();
+        if (m_backgroundTaskInstance)
+        {
+            m_backgroundTaskInstance.Canceled(token);
+        }
+    }
+    bool PushNotificationReceivedEventArgs::Handled()
+    {
+        auto lock = m_lock.lock_shared();
+        if (m_args)
+        {
+            return m_args.Cancel();
+        }
+        return false;
+    }
+    void PushNotificationReceivedEventArgs::Handled(bool value)
+    {
+        auto lock = m_lock.lock_shared();
+        if (m_args)
+        {
+            m_args.Cancel(value);
+        }
```

Comment:
Is it an error to try to handle something that can't be handled?

Reply:
This check is because there is nothing stopping someone from calling Handled() in the case of background activation, which doesn't work because m_args is null.

Reply:
So it should be an error, not silently ignored.